### PR TITLE
Include capabilities in CSIAddonsNode status

### PR DESCRIPTION
**Pull request by black-dragon74**
This patch adds support to display advertised
capabilities in the status field of CSIAddonsNode

The capabilities are displayed in the format `capability.Type` like:
```yaml
Status:
  Capabilities:
    service.NODE_SERVICE
    reclaim_space.ONLINE
    encryption_key_rotation.ENCRYPTIONKEYROTATION
  Message:  Successfully established connection with sidecar
  State:    Connected
```

Closes: #376

### CODE REVIEW

File: internal/controller/csiaddons/csiaddonsnode_controller.go [hunk 1]
```diff
@@ -271,3 +273,21 @@ func validateCSIAddonsNodeSpec(csiaddonsnode *csiaddonsv1alpha1.CSIAddonsNode) e
 
 	return nil
 }
+
+// parseCapabilities returns a list of capabilities in the format
+// capability.Type
+func parseCapabilities(caps []*identity.Capability) []string {
```
Comment (Madhu-1) [inline internal/controller/csiaddons/csiaddonsnode_controller.go:279]: add unit test for this one?

Comment (black-dragon74) [reply internal/controller/csiaddons/csiaddonsnode_controller.go:279]: Done.

File: internal/controller/csiaddons/csiaddonsnode_controller.go [hunk 2]
```diff
@@ -271,3 +273,21 @@ func validateCSIAddonsNodeSpec(csiaddonsnode *csiaddonsv1alpha1.CSIAddonsNode) e
 
 	return nil
 }
+
+// parseCapabilities returns a list of capabilities in the format
+// capability.Type
```
Comment (nixpanic) [inline internal/controller/csiaddons/csiaddonsnode_controller.go:278]: can you add a little note/example of how a `cap.String()` looks like, and what the desired output for it is?

File: internal/controller/csiaddons/csiaddonsnode_controller.go [hunk 3]
```diff
@@ -271,3 +273,21 @@ func validateCSIAddonsNodeSpec(csiaddonsnode *csiaddonsv1alpha1.CSIAddonsNode) e
...
+	if len(caps) == 0 {
+		return []string{}
+	}
+
+	capabilities := make([]string, 0, len(caps))
+
+	for _, cap := range caps {
+		capStr := strings.ReplaceAll(cap.String(), ":{type:", ".")
+		capStr = strings.ReplaceAll(capStr, "}", "")
+		capabilities = append(capabilities, capStr)
```
Comment (nixpanic) [inline internal/controller/csiaddons/csiaddonsnode_controller.go:289]: you already created a right sized slice, no need to re-create a new `capabilities` slice in every loop. You can be a little more efficient like this:

```go
for i, cap := range caps {
    ...
    capabilities[i] = capStr
}
```

Comment (black-dragon74) [reply internal/controller/csiaddons/csiaddonsnode_controller.go]: Added a fixed slice later as an optimization and forgot to update the rest of the code to actually use that optimization. Classic.

Thank you! Fixed :)

File: internal/controller/csiaddons/csiaddonsnode_controller.go (hunk 2 above)
Comment (black-dragon74) [reply internal/controller/csiaddons/csiaddonsnode_controller.go:278]: Done

**Review by nixpanic — APPROVED** [review]

**Review by Madhu-1 — APPROVED** [review]

**Comment by nixpanic** [comment]
@mergifyio requeue

**Comment by nixpanic** [comment]
@mergifyio rebase

**Pull request merged**